k1run.sh

```diff
@@ -0,0 +1,176 @@
+    echo "Invalid number of arguments passed in. Must be between 2 and 3 arguments"
+    help_messages
+    return
+fi
+
+cwd=$( dirname "$PWD" )
+
+folder="${1}"
+
+folder="${cwd}/${folder}"
+if [ ! -d "${folder}" ]; then
+   echo "${folder} is not valid"
+   return
+fi
+
+SWITCH=$2
+SWITCH2=$3
+
+# Variables
+# MOODLE_DOCKER_DB - database used by Moodle - default maria db.
+# MOODLE_DOCKER_WWWROOT - folder where the Moodle code is located;
+# MOODLE_DOCKER_PHP_VERSION - version of PHP used by Moodle. Default 8.1.
+
+if [ "$SWITCH" = "--help" ]; then
+    help_messages
+fi
+
+# Check to see if the options are valied.
+
+list_of_options="--build --down --destroy --reboot --load --phpunit --behat"
+
+if  exists_in_list "$list_of_options" " " $SWITCH;  then
+    echo "Invalid option $SWITCH"
+    help_messages
+    return
+fi
+
+if [ "$variablecount" -eq 3 ]; then
+    if  exists_in_list "$list_of_options" " " $SWITCH2;  then
+        echo "Invalid option $SWITCH2"
+        help_messages
+        return
+    fi
+fi
+
+# Always use mariadb as a database.
+
+export MOODLE_DOCKER_DB=mariadb
+export MOODLE_DOCKER_WWWROOT=${folder}
+
+# Check the Moodle version. If its 4.5 then set php version to 8.3
+moodlever=$(grep "$branch   = '405';" $folder/version.php)
+if [ "$moodlever" ]; then
+   export MOODLE_DOCKER_PHP_VERSION=8.3
+fi
+
+# Use the local.yml_single for one site - includes adminer.
+cp local.yml_single local.yml
+cp config.docker-template.php $MOODLE_DOCKER_WWWROOT/config.php
+
+# Build
+if [ "$SWITCH" = "--build" ]; then
+    # Start up containers
+    bin/moodle-docker-compose up -d
+    # Wait for DB to come up
+    bin/moodle-docker-wait-for-db
+    # Initialize the database
+    bin/moodle-docker-compose exec webserver php admin/cli/install_database.php --agree-license --fullname="K1MOODLE" --shortname="K1MOODLE" --summary="K1 Moodle dev" --adminpass="test" --adminemail="admin@example.com"
+    if [ "$SWITCH2" = "--phpunit" ]; then
+       # Add in unit tests initialization.
+       bin/moodle-docker-compose exec webserver php admin/tool/phpunit/cli/init.php
+    fi
+    if [ "$SWITCH2" = "--behat" ]; then
+       # Add in unit tests initialization.
+       bin/moodle-docker-compose exec webserver php admin/tool/behat/init.php
+    fi
+    adminer_plugins
+fi
+
+# DESTROY
+if [ "$SWITCH" = "--destroy" ]; then
+    bin/moodle-docker-compose down
+fi
+
+# PNPUNIT ONLY
+if [ "$SWITCH" = "--phpunit" ]; then
+   # Start up containers
+   bin/moodle-docker-compose up -d
+   # Wait for DB to come up
+   bin/moodle-docker-wait-for-db
+   bin/moodle-docker-compose exec webserver php admin/tool/phpunit/cli/init.php
+   adminer_plugins
+fi
+
+# BEHAT ONLY
+if [ "$SWITCH" = "--behat" ]; then
+   # Start up containers
+   bin/moodle-docker-compose up -d
+   # Wait for DB to come up
+   bin/moodle-docker-wait-for-db
+   bin/moodle-docker-compose exec webserver php admin/tool/behat/cli/init.php
+fi
+
+# REBOOT
+if [ "$SWITCH" = "--reboot" ]; then
```

Comment:
Do we have to reinstall Moodle on a reboot?

Reply:
Nice catch - updated the code to stop and start the containers.